master/internal/trial_workload_sequencer.go

```diff
@@ -221,6 +224,9 @@ func (s *trialWorkloadSequencer) computeValidationMetricsCompleted(
 	msg searcher.CompletedMessage, isBestValFuture actor.Response,
 ) (searcher.Runnable, interface{}) {
 	s.batchesSinceLastVal = 0
+	if s.needInitialValidation {
+		s.needInitialValidation = false
```

Comment:
question: not very familiar with how we save the state of the workload sequencer, but what prevents `s.needInitialValidation` from being true every time a trial restarted with a master restart?

Reply:
Never mind just realized this would likely be handled by replaying the event log right?

Reply:
Yeah, it will be true on a restart but if the validation was completed it will be replayed from the flushed searcher events. I made a change to not rollback any events to do with the zeroth step; since we never train any in it and it must start from a checkpoint, we can assume anything that happened is still valid (but for now anything is only validations).

Reply:
Yeah saw that and that makes sense to do, can't think of a good reason that we would want to re-run that initial validation.